### ISSUE TITLE
Move hydra.ini to config.py

### ISSUE THREAD

**Issue by knoxsp** [hydraplatform/hydra-base#98]
Rather than use ini files, move to using config.py. This closely matches the method used by flask, and as such makes for a more coherent and consistent config management.

**Comment by rheinheimer** [comment]
Good idea. Especially as you can then use config classes, and load a different config class depending on the installation type.

What about using environment variables instead / as well?  I added something to my fork to map environment variables to hydra.ini variables. I don't recall the exact scheme, but it was something like `HYDRA__[SECTION]__[OPTION]=[VALUE]`.

I also use [python-dotenv](https://pypi.org/project/python-dotenv/), which could be useful.
